pageserver/src/tenant.rs

```diff
@@ -2031,7 +2032,13 @@ impl Tenant {
             pg_version,
             initial_logical_size_can_start.cloned(),
             initial_logical_size_attempt.cloned(),
-        ))
+        );
+
+        if matches!(cause, CreateTimelineCause::Delete) {
+            timeline.set_state(TimelineState::Stopping);
+        }
```

Comment:
Why not push this down further into `Timeline::new` ?

Reply:
See updated version

Reply:
Updated version got reviewed, resolving